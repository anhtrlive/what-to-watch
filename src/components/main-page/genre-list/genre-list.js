import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

class GenreList extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {genre, films, clickHandlerGenre, resetShowCard} = this.props;
    let allGennre = new Set(films.map((el) => el.genre));
    allGennre = [...allGennre.keys()];
    allGennre.sort();
    allGennre = [`All genres`, ...allGennre];

    return <ul className="catalog__genres-list">
      {allGennre.map((element, index) => {
        const liClasses = (genre === element) ?
          `catalog__genres-item catalog__genres-item--active` :
          `catalog__genres-item`;

        return <li
          className ={liClasses}
          onClick={(evt) => {
            clickHandlerGenre(evt);
            resetShowCard();
          }}
          key={`${element}-${index}`}
        >
          <a href="#a" className="catalog__genres-link">
            {element}
          </a>
        </li>;
      })}
    </ul>;
  }
}

GenreList.propTypes = {
  genre: PropTypes.string.isRequired,
  films: PropTypes.array.isRequired,
  clickHandlerGenre: PropTypes.func.isRequired,
  resetShowCard: PropTypes.func.isRequired,
};

export {GenreList};
