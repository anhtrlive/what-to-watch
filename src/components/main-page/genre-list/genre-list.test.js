import React from 'react';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {BrowserRouter} from 'react-router-dom';
import {GenreList} from './genre-list';
import {films} from '../../../mocks/films';

Enzyme.configure({adapter: new Adapter()});

const genre = films[0].genre;
const clickHandlerGenre = jest.fn();
const resetShowCard = jest.fn();

it(`button plus list`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <GenreList
          films={films}
          genre={genre}
          clickHandlerGenre={clickHandlerGenre}
          resetShowCard={resetShowCard}
        />
      </BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
