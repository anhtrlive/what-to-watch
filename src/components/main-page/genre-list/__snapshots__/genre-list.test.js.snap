// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button plus list 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#a"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#a"
    >
      Comedy
    </a>
  </li>
</ul>
`;
